### PR TITLE
#195 Connect type filter

### LINKED ISSUE #195: Connect the type filter
The types have modified in the `data.json` which is in this repo: https://github.com/sussol/universal-drug-codes-server/

After importing, there is only one `medicinal_product` - but there are `drugs`
Would be nice to wire up the type filter now.

## Expected Behaviour
By default, `drugs` is selected. Clicking one of the other options will display the relevant results.

### PR DESCRIPTION
**Pull request by mark-prins**
Fixes #195 

## Description
Connects the type toggle to the variables in redux ( which is about to be refactored ).
Allows filtering of results by toggle. To see this in action - import the updated data, helpfully saved for you in `import.json` in the `/tools/scripts/dgraph/data/` folder

Also have changed the default from `medicinal_product` to `drug` as discussed early on in the project.

### Documentation:

Tick below if one of the following applies:

1. Documentation has been added to cover changes introduced by this PR.
2. This PR requires no changes to documentation.

- [x] Documentation OK.

### Tests:

Tick below if one of the following applies:

1. Tests have been added to cover changes introduced by this PR.
2. This PR requires no new tests.

- [ ] Tests OK.

### CODE REVIEW

**Review by katherine-sussol — APPROVED** [review]
Nice! Looks good to me.
I have a few minor usability thoughts (for edge cases) if there's time later (but the basic functionality works well!) I'm also no usability expert - just one of those annoying users that intentionally tries to break the UI 😈 :
- Unselecting the last tab could maybe return the filter to default state of 'Drugs' selected? (since this is what the Explorer does)
- Paging through something with lots of results and then filtering to something with less (e.g. page 10 of `Drug` with no search term -> then click `Other` and unselecting `Drug`) gives the user an empty table with something like `126-1 of 1.` Is it worth doing something like resetting the paging on filter change?

File: unified-codes/libs/ui/src/lib/components/entity/organisms/EntityBrowser.tsx [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import * as React from 'react';
 
-import { IEntityCollection, IExplorerVariables } from '@unified-codes/data';
+import { IEntityCollection, IEntityType, IExplorerVariables } from '@unified-codes/data';
```
[the remark this answers is not in the document]
Comment (mark-prins) [reply unified-codes/libs/ui/src/lib/components/entity/organisms/EntityBrowser.tsx:3]: oops - and odd that I didn't spot it. thanks!

File: unified-codes/package-lock.json [hunk 2]
```diff
@@ -13483,6 +13493,13 @@
         }
       }
     },
+    "file-uri-to-path": {
```
[the remark this answers is not in the document]
Comment (mark-prins) [reply unified-codes/package-lock.json:13496]: there was a requirement for `reselect` in one of the selectors. but I didn't have that in the `package.json` so it wasn't compiling 🤷‍♂️

**Comment by mark-prins** [comment]
> I have a few minor usability thoughts

Thanks @katherine-sussol and well spotted. I agree, and usability is important!
I've addressed these in 4f1fb343a - with the 'unselecting all' one, I've gone with option (b) and shown no results. For me, this matches logically with what I've done ( by un-selecting all types ). See how it feels for you though.

**Comment by katherine-sussol** [comment]
> with the 'unselecting all' one, I've gone with option (b) and shown no results. For me, this matches logically with what I've done ( by un-selecting all types ). See how it feels for you though.

Oh, that works nicely! Upon reflection, unselecting everything and then suddenly seeing the filter have a mind of its own might startle me 🙀  - so I think showing nothing is more intuitive.
